shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API/Validation.hs

```diff
@@ -67,6 +72,44 @@ class
   ) =>
   ApplyBlock era
   where
+
+  -- | Apply the header level ledger transition.
+  --
+  -- This handles checks and updates that happen on a slot tick, as well as a
+  -- few header level checks, such as size constraints.
```

Comment:
so this is just `TICK`, but where you can supply your own options, right? Maybe a comment to that effect?